### DISCUSSION TITLE
Allow extension/use of ignore patterns (in files) via configuration

### DISCUSSION THREAD

**Discussion by ghost** [eslint/eslint#16441] [Ideas]
I currently am copying my .gitignore file and adding a few other patterns/files to my .gitlintignore. It would be great to specify somewhere in the configs to say that it should use .gitignore for ignoring certain patterns. It would be even better if I could extend .eslintignore from .gitignore

**Comment by mdjermanovic** [comment]
You can load .gitignore file into [`ignorePatterns`](https://eslint.org/docs/latest/user-guide/configuring/ignoring-code#ignorepatterns-in-config-files) array in your `.eslintrc.js` config file, and then add additional patterns.

```js
// .eslintrc.js

const fs = require("fs");

module.exports = {
    ignorePatterns: [
        ...fs.readFileSync(path.join(__dirname, ".gitignore"), "utf8")
            .split("\n")
            .filter(line => line && !line.startsWith("#")),
        // "additional pattern 1",
        // "additional pattern 2",
        // ....
    ]
};
```
